qemu/tests/qemu_guest_agent.py

```diff
@@ -1918,8 +1946,129 @@ def gagent_check_fsfreeze(self, test, params, env):
                 logging.warn("Finally failed to thaw guest fs,"
                              " detail: '%s'", detail)
             raise
+        # check after fsthaw
+        self._action_after_fsthaw(write_cmd_guest, write_cmd_timeout)
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze(self, test, params, env):
+        """
+        Test guest agent commands "guest-fsfreeze-freeze"
+
+        Test steps:
+        1) Check the FS is thawed.
+        2) Freeze the FS.
+        3) Check the FS is frozen from both guest agent side and
+         guest os side.
+        4) Thaw the FS.
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        self._fsfreeze()
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze_list(self, test, params, env):
+        """
```

Comment:
@xiagao you have added command guest-fsfreeze-freeze-list to avocado-vt. why no use it directly ?

Reply:
There is no guest-fsfreeze-freeze-list in avocado-vt. only fsfreeze() function.

Reply:
I saw your avocado-vt's patch about command guest-fsfreeze-freeze-list.
.........
        if fsfreeze_list:
            cmd = "guest-fsfreeze-freeze-list"
            if mountpoints:
                args = {"mountpoints": mountpoints}

        if self.check_has_command(cmd):
            ret = self.cmd(cmd=cmd, timeout=timeout)
            ret = self.cmd(cmd=cmd, timeout=timeout, args=args) 
........

I have a little confuse,  avocado-vt's command guest-fsfreeze-freeze-list also freeze mounted file system.   why cann't use it directly.  when will avocado-vt's guest-fsfreeze-list command be used?

Reply:
These code are in fsfreeze() function, I conclude  guest-fsfreeze-freeze-list  and guest-fsfreeze-freeze into one function, use parameter to decide use which cmd.

Reply:
To my understanding,  you don't need to re-define this function.  If use avocado-vt's command guest-fsfreeze-freeze-list and guest-fsfreeze-freeze to develop this case, It is easier. And you mentioned "depends on: avocado-framework/avocado-vt#2499" in comments.  If need, we can discuss this issue further later via IRC.

Reply:
xiangchun, in  avocado-framework/avocado-vt#2499 ，there is no guest-fsfreeze-freeze-list function

Reply:
@xiagao why add guest-fsfreeze-freeze-list command to avocado-vt? what is it's function if I call command guest-fsfreeze-freeze-list from avocado-vt?

Reply:
you are right,  you put command to function in avocado-vt.  my unstanding is wrong.